### DISCUSSION TITLE
Updating datasource from git repo fails: "No module named 'dulwich'"

### DISCUSSION THREAD

**Discussion by fallenfuzz** [netbox-community/netbox#17780] [Help Wanted!]
### Deployment Type

Self-hosted

### Triage priority

N/A

### NetBox Version

4.1.4

### Python Version

3.10

### Steps to Reproduce

Configure a new data source of the type "Git"
Update the data source by clicking the "Sync" button

### Expected Behavior

The job should complete without errors.

### Observed Behavior

The job fails with the following error:

SyncError("There was an error initializing the backend. A dependency needs to be installed: No module named 'dulwich'")
The issue can be worked-around by entering the venv and installing the module 'dulwich' manually with:

pip install dulwich

**Comment by candlerb** [comment]
> The issue can be worked-around by entering the venv and installing the module 'dulwich' manually

If that error message isn't clear enough: you need to run

```
/opt/netbox/venv/bin/pip install dulwich
```

(possibly as root) to fix the problem. To make the solution permanent (including through future upgrades), edit or create `/opt/netbox/local_requirements.txt`, add the line `dulwich`.

The documentation also mentions this [here](https://netboxlabs.com/docs/netbox/en/stable/features/synchronized-data/) although without telling you exactly *how* to install the package.

**Comment by fallenfuzz** [comment]
@candlerb I fixed it, yes and local_requirements is also a way to fix, but shouldn't it be included?
It is not customer plugin, but part of the system no?

I see that from v3.6.0 (2023-08-30) it isn't installed by default, will include in local_requirements then.

The boto3 and dulwich packages are no longer installed automatically. If needed for S3/git remote data backend support, add them to local_requirements.txt to ensure their installation.
